### DISCUSSION TITLE
in TailwindCss3 with (darkMode:'class'), in light mode, "(prefers-color-scheme: dark)" classes is applied and show dark

### DISCUSSION THREAD

**Discussion by nekooee** [tailwindlabs/tailwindcss#6462] [Help]
Hi,
I added ```darkMode: 'class'``` in config and change light and dark like below code:

```javascrip
//switch light nd dark mode
window.darkThemeSwitch = function () {
    if (localStorage.theme === 'dark') {
        localStorage.theme = 'light'
        document.documentElement.classList.remove('dark')
    } else {
        localStorage.theme = 'dark'
        document.documentElement.classList.add('dark')
    }
}

document.addEventListener("DOMContentLoaded", function () {
    if (localStorage.theme === 'dark' || (!('theme' in localStorage) && window.matchMedia('(prefers-color-scheme: dark)').matches)) {
        document.documentElement.classList.add('dark')
    } else {
        document.documentElement.classList.remove('dark')
    }
});
```

my html code:
```html
<div class="text-gray-500 dark:text-gray-400">sample text</div>
```

But in the light state, the dark class still applies:
![myScreenShot](https://user-images.githubusercontent.com/6003803/145837416-ebbe0ce6-2408-4346-ae14-8f005562c8dc.png)

please help me. thank you.

**Comment by NovusOpus** [comment]
Did you also added the dark class to your opening `<html>` tag to enable Dark Mode?

You can read more about it: [Toggling dark mode manually - Tailwind CSS Docs](https://tailwindcss.com/docs/dark-mode#toggling-dark-mode-manually)

**Reply by nekooee** [reply]
> Did you also added the `dark` class to your opening `<html>` tag to enable Dark Mode?
> 
> You can read more about it: [Toggling dark mode manually - Tailwind CSS Docs](https://tailwindcss.com/docs/dark-mode#toggling-dark-mode-manually)

thank you.
Yes, I add dark in the night mode. But now in the light state, despite the dark being removed, the night class is still applied. The photo below can better show my problem:

![2021-12-14 14_03_39-Settings](https://user-images.githubusercontent.com/6003803/145982596-278a2dd4-15dd-4efe-a8ca-048c01218bb1.png)
![2021-12-14 14_04_38-Settings](https://user-images.githubusercontent.com/6003803/145982607-a8b08252-cf20-4429-af7c-56c36fab0a86.png)

**Comment by nekooee** [comment]
@adamwathan is this bug??

**Comment by nekooee** [comment]
I found the problem. This problem occurs when I use tailwind CDN. And when the site is published, when I deactivate the CDN, everything goes well. I even found other problems that arise when activating a CDN.

**Comment by nekooee** [comment]
When is this CDN problem solved? This is really annoying. None of the TailwindCSS programmers even commented on this...

**Reply by adamwathan** [reply]
If you've found a bug, open an issue. We don't monitor discussions closely, they are for the community.

**Comment by usmanlogin** [comment]
In my case I was adding darkMode: 'class', to the theme object, It should be added in the module.export object.
